Sources/Parsers/AssetsCatalogParser.swift

```diff
@@ -34,18 +34,48 @@ extension Catalog.Entry: Equatable {
 }
 
 extension Catalog.Entry: Comparable {
-  static func <(lhs: Catalog.Entry, rhs: Catalog.Entry) -> Bool {
+  static func < (lhs: Catalog.Entry, rhs: Catalog.Entry) -> Bool {
     switch (lhs, rhs) {
-    case (.group(let lhsName, _), .group(let rhsName, _)):
+    case (.group(let lhsName, let lhsItems), .group(let rhsName, let rhsItems)):
+      if lhsName == rhsName {
+        return compare(lhsItems, rhsItems) == .orderedAscending
+      }
       return lhsName < rhsName
-    case (.color(let lhsName, _), .color(let rhsName, _)):
+    case (.color(let lhsName, let lhsValue), .color(let rhsName, let rhsValue)):
+      if lhsName == rhsName {
+        return lhsValue < rhsValue
+      }
       return lhsName < rhsName
-    case (.image(let lhsName, _), .image(let rhsName, _)):
+    case (.image(let lhsName, let lhsValue), .image(let rhsName, let rhsValue)):
+      if lhsName == rhsName {
+        return lhsValue < rhsValue
+      }
       return lhsName < rhsName
     default:
       return false
     }
   }
+
+  private static func compare(_ lhs: [Catalog.Entry], _ rhs: [Catalog.Entry]) -> ComparisonResult {
+    guard let lhsFirst = lhs.first else {
+      if rhs.first == nil {
+        return .orderedSame
+      }
+      return .orderedAscending
+    }
+
+    guard let rhsFirst = rhs.first else {
+      return .orderedDescending
+    }
+
+    if lhsFirst < rhsFirst {
+      return .orderedAscending
+    } else if lhsFirst > rhsFirst {
+      return .orderedDescending
+    }
+
+    return compare(Array(lhs.dropFirst()), Array(rhs.dropFirst()))
```

Comment:
I'm not a fan of using this recursive approach as it can make the stack grow quite quickly for a group containing 100s of images…

What about:
```swift
private static func compare(_ lhs: [Catalog.Entry], _ rhs: [Catalog.Entry]) -> ComparisonResult {
  for pair in zip(lhs, rhs) {
    if pair.0 < pair.1 { return .orderedAscending }
    if pair.0 > pair.1 { return .orderedDescending }
  }
  // If we arrive a this point, this means all pairs have been considered equal
  // but one of the list might be longer than the other
  if lhs.count < rhs.count { return .orderedAscending }
  if lhs.count > rhs.count { return .orderedDescending }
  return .orderedSame
}
```

Reply:
That's a good point. Updating 👍

Reply:
I even wonder if the last part (last 3 lines) couldn't be reduced to `return compare(lhs.count, rhs.count)`?
(Don't we have a `compare(Int,Int) -> ComparisonResult` in the stdlib?)